prefect_shell/commands.py

```diff
@@ -94,7 +93,9 @@ def example_shell_run_command_flow():
                 )
                 raise RuntimeError(msg)
         
-        if os.path.exists(tmp.name):
+        try:
+            os.path.exists(tmp.name)
```

Comment:
I think anything wrapped in the `NamedTemporaryFile` context manager should be under try so a file isn't left behind if something fails in there.

If that's too much indentation, to reduce indentation levels, we can use NamedTemporaryFile as a normal statement, e.g.:
`tmp = tempfile.NamedTemporaryFile(prefix="prefect-", suffix=extension, delete=False)`

Reply:
Yeah, I've come around to getting rid of the context manager, so I've pulled it out as you described.